src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -80,10 +80,17 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '' or '$(TargetFrameworkVersion)' == ''">
     <_UnsupportedTargetFrameworkError>true</_UnsupportedTargetFrameworkError>
   </PropertyGroup>
-  <Target Name="_CheckForUnsupportedTargetFramework" BeforeTargets="_CheckForInvalidConfigurationAndPlatform">
-    <NETSdkError Condition="'$(_UnsupportedTargetFrameworkError)' == 'true'" 
+  <Target Name="_CheckForUnsupportedTargetFramework"
+          BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences"
```

Comment:
The message from restore is:

> error : Invalid restore input. Invalid target framework 'unsupported'. Input files: C:\git\dotnet-sdk\bin\repro\WrongFramework\WrongFramework.csproj.

Note that it says that the target framework is "unsupported" for any invalid value of `TargetFramework`.

Also, in the case that you have multiple valid target frameworks separated by semicolons in the `TargetFramework` property, the restore operation appears to just work, restoring for both.